### PR TITLE
fix: return errors on CreateSecret call

### LINKED ISSUE #655: pipeline init can't create secret if secret is scheduled for deletion
Was running into issues where running `pipeline init` on a project for which there was a previous pipeline and pipeline secret would silently succeed without creating the secret because of the following error from the call to `SecretsManager.CreateSecret()`:

```
InvalidRequestException: You can't create this secret because a secret with this name is already scheduled for deletion.
```

This is bc this error is not properly surfaced/handled in [secretsmanager.go](https://github.com/aws/amazon-ecs-cli-v2/blob/1a91c0ea9a5025e6b9ff85a109cee06efb8350b4/internal/pkg/store/secretsmanager/secretsmanager.go#L48-L57).

### PR DESCRIPTION
**Pull request by SoManyHs**
Also adds missing tests for secretsmanager
Fixes #655.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by SoManyHs** [comment]
Previous behavior:
```
~/aws-reinvent-2019-trivia-backend(master)$ ecs-preview pipeline init

? Would you like to add an environment to your pipeline? Yes

? Which environment would you like to add to your pipeline? prod

? Would you like to add another environment to your pipeline? Yes

? Which environment would you like to add to your pipeline? test

? Which GitHub repository would you like to use for your application? git@github.com:SoManyHs/aws-reinvent-trivia-2019-backend

? Please enter your GitHub Personal Access Token for your repository: aws-reinvent-trivia-2019-backend [? for help] ********************
✔ Wrote the pipeline manifest for aws-reinvent-trivia-2019-backend at 'ecs-project/pipeline.yml'
✔ Wrote the buildspec for the pipeline's build stage at 'ecs-project/buildspec.yml'
The manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.
The buildspec contains the commands to build and push your container images to your ECR repositories.
```

but the secret was not actually being created; debug message added in the code gave:
```
ERROR CREATING SECRET: InvalidRequestException: You can't create this secret because a secret with this name is already scheduled for deletion.

CREATED SECRET: {

}
```

With this change, behavior is now:
```
~/aws-reinvent-2019-trivia-backend(master)$ ecs-preview pipeline init

? Would you like to add an environment to your pipeline? Yes

? Which environment would you like to add to your pipeline? prod

? Would you like to add another environment to your pipeline? No

? Which GitHub repository would you like to use for your application? git@github.com:SoManyHs/aws-reinvent-trivia-2019-backend

? Please enter your GitHub Personal Access Token for your repository: aws-reinvent-trivia-2019-backend [? for help] *****
Error: create secret github-token-backend-aws-reinvent-trivia-2019-backend: InvalidRequestException: You can't create this secret because a secret with this name is already scheduled for deletion.
```

File: internal/pkg/cli/pipeline_init.go [hunk 1]
```diff
@@ -181,6 +181,7 @@ func (o *initPipelineOpts) Execute() error {
 		}
 		log.Successf("Secret already exists for %s! Do nothing.\n", color.HighlightUserInput(o.GitHubRepo))
 	}
+	log.Successf("Created the secret %s for pipeline source stage\n", color.HighlightUserInput(secretName))
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: nit: Can we end the log message with a punctuation? [STYLE_GUIDE#use full-sentences](https://github.com/aws/amazon-ecs-cli-v2/blob/master/STYLE_GUIDE.md#use-full-sentences)

Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: Are we going to log success twice for a secret now if the secret already exists?

File: internal/pkg/store/secretsmanager/secretsmanager.go [hunk 2]
```diff
@@ -56,6 +56,8 @@ func (s *SecretsManager) CreateSecret(secretName, secretString string) (string,
 				}
 			}
 		}
+		return "", fmt.Errorf("create secret %s: %w", secretName, err)
```
Comment (efekarakus) [inline internal/pkg/store/secretsmanager/secretsmanager.go]: 👏

**Comment by SoManyHs** [comment]
@iamhopaul123 Yes, we do that when we use our own secretsmanager client elsewhere (e.g. pipeline_init.go) and have [mocks for that already](https://github.com/aws/amazon-ecs-cli-v2/blob/master/mocks/mock_secret.go). However, in this case I believe we are trying to test the behavior of what the service call is doing itself -- we currently do this for the [IAM](https://github.com/aws/amazon-ecs-cli-v2/blob/master/Makefile#L149), though we've been writing the mocks by hand elsewhere (see: [cloudformation_test.go](https://github.com/aws/amazon-ecs-cli-v2/blob/master/internal/pkg/deploy/cloudformation/cloudformation_test.go). Changes to the interface upstream won't affect us unless we re-generate our dependencies, IIRC.

File: internal/pkg/cli/pipeline_init.go (hunk 1 above)
Comment (SoManyHs) [reply internal/pkg/cli/pipeline_init.go]: Oh good call. I can fix that.

**Comment by SoManyHs** [comment]
With ec4d058:
```
~/aws-reinvent-2019-trivia-backend(master)$ ecs-preview pipeline init

? Would you like to add an environment to your pipeline? Yes

? Which environment would you like to add to your pipeline? prod

? Would you like to add another environment to your pipeline? No

? Which GitHub repository would you like to use for your application? git@github.com:SoManyHs/aws-reinvent-trivia-2019-backend

? Please enter your GitHub Personal Access Token for your repository: aws-reinvent-trivia-2019-backend [? for help] ***
✔ Secret already exists for aws-reinvent-trivia-2019-backend! Do nothing.
✔ Wrote the pipeline manifest for aws-reinvent-trivia-2019-backend at 'ecs-project/pipeline.yml'
✔ Wrote the buildspec for the pipeline's build stage at 'ecs-project/buildspec.yml'
The manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.
The buildspec contains the commands to build and push your container images to your ECR repositories.

```

File: internal/pkg/cli/pipeline_init.go (hunk 1 above)
Comment (SoManyHs) [reply internal/pkg/cli/pipeline_init.go]: resolved by ec4d058

**Comment by efekarakus** [comment]
> @iamhopaul123 Yes, we do that when we use our own secretsmanager client elsewhere (e.g. pipeline_init.go) and have [mocks for that already](https://github.com/aws/amazon-ecs-cli-v2/blob/master/mocks/mock_secret.go). However, in this case I believe we are trying to test the behavior of what the service call is doing itself -- we currently do this for the [IAM](https://github.com/aws/amazon-ecs-cli-v2/blob/master/Makefile#L149), though we've been writing the mocks by hand elsewhere (see: [cloudformation_test.go](https://github.com/aws/amazon-ecs-cli-v2/blob/master/internal/pkg/deploy/cloudformation/cloudformation_test.go)). Changes to the interface upstream won't affect us unless we re-generate our dependencies, IIRC.

I think what we do for IAM is probably bad :D (blame young Efe).

If IAM or SecretsManager adds a new method then unless we run `gen-mocks` again the mock won't satisfy the `iamiface` anymore and the build will fail :(

Here is an example that gets around it by creating an interface of only the methods needed from the `iface`:
https://github.com/aws/amazon-ecs-cli-v2/blob/b223142e566e3bd123b541ff8cc500d4b3a113fb/internal/pkg/aws/ecs/ecs.go#L14-L21

**Pull request merged**